metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/EnrichmentWorkerImpl.java

```diff
@@ -69,86 +55,103 @@
   }
 
   @Override
-  public byte[] process(InputStream inputStream) throws DereferenceOrEnrichException {
-    return process(inputStream, Mode.DEREFERENCE_AND_ENRICHMENT);
+  public byte[] process(InputStream inputStream)
+      throws EnrichmentException, DereferenceException, SerializationException {
+    return process(inputStream, new HashSet<>(Arrays.asList(Mode.values())));
   }
 
   @Override
-  public byte[] process(final InputStream inputStream, Mode mode)
-      throws DereferenceOrEnrichException {
+  public byte[] process(final InputStream inputStream, Set<Mode> modes)
+      throws SerializationException, EnrichmentException, DereferenceException {
     if (inputStream == null) {
       throw new IllegalArgumentException("The input stream cannot be null.");
     }
     try {
       final RDF inputRdf = convertInputStreamToRdf(inputStream);
-      final RDF resultRdf = process(inputRdf, mode);
+      final RDF resultRdf = process(inputRdf, modes);
       return convertRdfToBytes(resultRdf);
     } catch (JiBXException e) {
-      throw new DereferenceOrEnrichException(
+      throw new SerializationException(
           "Something went wrong with converting to or from the RDF format.", e);
+    } catch (EnrichmentException e){
+      throw new EnrichmentException(
+          "Something went wrong with the enrichment from the RDF file.", e);
+    } catch (DereferenceException e){
+      throw new DereferenceException(
+          "Something went wrong with the dereference from the RDF file.", e);
     }
   }
 
   @Override
-  public String process(String inputString) throws DereferenceOrEnrichException {
-    return process(inputString, Mode.DEREFERENCE_AND_ENRICHMENT);
+  public String process(String inputString)
+      throws EnrichmentException, DereferenceException, SerializationException {
+    return process(inputString, new HashSet<>(Arrays.asList(Mode.values())));
```

Comment:
Perhaps this is cleaner:
EnumSet.allOf(Mode.class);

Reply:
Done